src/viam/sdk/services/motion/server.cpp

```diff
@@ -103,37 +98,18 @@ ::grpc::Status MotionServer::GetPose(
     ::grpc::ServerContext* context,
     const ::viam::service::motion::v1::GetPoseRequest* request,
     ::viam::service::motion::v1::GetPoseResponse* response) noexcept {
-    if (!request) {
-        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
-                              "Called [GetPose] without a request");
-    };
-
-    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
-    if (!rb) {
-        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
-    }
-
-    const std::shared_ptr<Motion> motion = std::dynamic_pointer_cast<Motion>(rb);
-
-    const auto& component_name = Name::from_proto(request->component_name());
-    const std::string& destination_frame = request->destination_frame();
-    std::vector<WorldState::transform> supplemental_transforms;
-    for (const auto& proto_transform : request->supplemental_transforms()) {
-        supplemental_transforms.push_back(WorldState::transform::from_proto(proto_transform));
-    }
-    AttributeMap extra;
-    if (request->has_extra()) {
-        extra = struct_to_map(request->extra());
-    }
-
-    const pose_in_frame pose = motion->get_pose(std::move(component_name),
-                                                std::move(destination_frame),
-                                                std::move(supplemental_transforms),
-                                                std::move(extra));
-
-    *response->mutable_pose() = pose.to_proto();
-
-    return ::grpc::Status();
+    return make_service_helper<Motion>(
+        "MotionServer::GetPose", this, request)([&](auto& helper, auto& motion) {
+        const auto& component_name = Name::from_proto(request->component_name());
+        const std::string& destination_frame = request->destination_frame();
+        std::vector<WorldState::transform> supplemental_transforms;
+        for (const auto& proto_transform : request->supplemental_transforms()) {
+            supplemental_transforms.push_back(WorldState::transform::from_proto(proto_transform));
+        }
+        const pose_in_frame pose = motion->get_pose(
+            component_name, destination_frame, supplemental_transforms, helper.getExtra());
+        *response->mutable_pose() = pose.to_proto();
+    });
```

Comment:
(flyby) Somehow missed this one when converting everything over to using `service_helper`.

Reply:
Nice catch!